### PR TITLE
Sync payment requests/responses for Klarna/Giropay/Ideal

### PR DESCRIPTION
**Pull request by kevin-reece-cko**
Sync'd Klarna/Giropay/Ideal (as new Ideal):
- Payment request source
- Payment response source (for get response and receive event)

Branch instance: https://api-reference.checkout.com/preview/task/sync-get-payment-responses/

The PR also contains the `sync-generated-specs.js` script to enable simple re-syncing of APM sources, and simple adding of new APMs. 
This script is run manually using `npm run sync-generated-specs`. Therefore reviewing the script is less important than reviewing the `yaml` changes made by the script.

### CODE REVIEW

File: spec/components/schemas/Payments/RequestSources/14_PaymentRequestKlarnaSource.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      - billing_address
+      - tax_amount
+      - products
+    properties:
+      authorization_token:
+        type: string
+        description: 'Klarna authentication token, obtained by the merchant during client transaction authorization.'
+      locale:
+        type: string
+        description: 'RFC 1766 customer''s locale. Optional, default taken from business settings.'
```
Comment (philippe-leonhardt-cko) [inline spec/components/schemas/Payments/RequestSources/14_PaymentRequestKlarnaSource.yaml]: Maybe stick with one string denotation:
- `" 'quote' "` double quotes around single quotes
- `' ''quote'' '` single quotes around escaped single quotes

File: spec/components/schemas/Payments/RequestSources/14_PaymentRequestKlarnaSource.yaml [hunk 2]
```diff
@@ -46,13 +46,13 @@ allOf:
         x-klarna-docs: 'https://developers.klarna.com/api/#payments-api__create-a-new-order__order_tax_amount'
       products:
         type: object
```
[the remark this answers is not in the document]
Comment (kevin-reece-cko) [reply spec/components/schemas/Payments/RequestSources/14_PaymentRequestKlarnaSource.yaml]: I don't think OpenAPI has typed arrays, so probably `array` is best
https://swagger.io/docs/specification/data-models/data-types/#array

File: spec/components/schemas/Payments/RequestSources/14_PaymentRequestKlarnaSource.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        x-cko-passthrough: true
+        x-klarna-docs: 'https://developers.klarna.com/api/#payments-api__create-a-new-order__order_lines'
+      customer:
+        type: object
+        description: 'Information about the liable customer of the order.'
+        x-klarna-docs: 'https://developers.klarna.com/api/#payments-api__create-a-new-order__customer'
+        x-cko-passthrough: true
+      merchant_reference1:
+        type: string
+        description: 'Used for storing merchant''s internal order number or other reference.  If set, will be shown on the confirmation page as "order number"  (max 255 characters).'
```
Comment (philippe-leonhardt-cko) [inline spec/components/schemas/Payments/RequestSources/14_PaymentRequestKlarnaSource.yaml]: remove extra whitespace

File: spec/components/schemas/Payments/RequestSources/14_PaymentRequestKlarnaSource.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        x-klarna-docs: 'https://developers.klarna.com/api/#payments-api__create-a-new-order__merchant_reference1'
+      merchant_reference2:
+        type: string
+        description: 'Used for storing merchant''s internal order number or other reference  (max 255 characters).'
```
Comment (philippe-leonhardt-cko) [inline spec/components/schemas/Payments/RequestSources/14_PaymentRequestKlarnaSource.yaml]: remove extra whitespace

File: scripts/sync-generated-specs.js [hunk 5]
```diff
@@ -47,6 +48,18 @@ var buildPaymentSource = function(requestData, paymentSourceName) {
   };
 };
 
+var addDescriptionToKlarnaPassthroughObjects = function(requestData, paymentSourceName) {
+  Object.keys(requestData.properties).forEach(propertyName => {
+    var property = requestData.properties[propertyName];
+    if (property['x-cko-passthrough'] === true && property['x-klarna-docs']) {
+      var apmPropertyName = property['x-klarna-name'] || propertyName;
+      var apmPropertyDocs = property['x-klarna-docs'];
+      property.description += '  \nThis object is passed directly to ' + paymentSourceName + ' as `' + apmPropertyName + '`, \
```
Comment (philippe-leonhardt-cko) [inline scripts/sync-generated-specs.js]: source to remove extra whitespace

File: spec/components/schemas/Payments/RequestSources/09_PaymentRequestIdealSource.yaml [hunk 6]
```diff
@@ -1,12 +1,26 @@
...
+      - bic
     properties:
-      issuer_id:
+      description:
+        maxLength: 35
         type: string
-        description: The issuer identifer
-        example: ABNANL2A
\ No newline at end of file
+        description: "Description of the product(s) or services being paid for.  This field must not contain characters that can lead to problems (for example those occurring in HTML editing codes).  To prevent any possible errors most iDEAL systems will reject any description that contains HTML-tags and such other code.\n"
```
Comment (philippe-leonhardt-cko) [inline spec/components/schemas/Payments/RequestSources/09_PaymentRequestIdealSource.yaml]: remove extra whitespace

File: spec/components/schemas/Payments/RequestSources/09_PaymentRequestIdealSource.yaml [hunk 7]
```diff
@@ -1,12 +1,26 @@
...
+      bic:
+        maxLength: 11
+        type: string
+        description: 'BIC (8 or 11-digits). In iDEAL-lingo this is also called issuerID'
+      language:
+        maxLength: 2
+        type: string
+        description: "This field enables the Issuer's site to select the Consumer's preferred language  (e.g. the language selected on the Merchant's site), if the Issuer's site supports this.  Code list in accordance with ISO 639-1. (Dutch = 'nl', English = 'en').  If a non-supported or non-existing language is entered the standard language of the Issuer is used. It is recommended to use 'nl' by default since not all Issuers support other languages.\n"
```
Comment (philippe-leonhardt-cko) [inline spec/components/schemas/Payments/RequestSources/09_PaymentRequestIdealSource.yaml]: remove extra whitespace

File: spec/components/schemas/Payments/ResponseSources/05_PaymentResponseIdealSource.yaml [hunk 8]
```diff
@@ -1,4 +1,26 @@
...
+    type: object
+    properties:
+      description:
+        maxLength: 27
+        type: string
+        description: description
+      bic:
+        maxLength: 11
+        type: string
+        description: "BIC (8 or 11-digits) BIC of the bank where the Consumer account is held.  If governing law prevents Issuers outside the Netherlands from disclosing this information, field may be omitted.\n"
```
Comment (philippe-leonhardt-cko) [inline spec/components/schemas/Payments/ResponseSources/05_PaymentResponseIdealSource.yaml]: remove extra whitespace

File: spec/components/schemas/Payments/ResponseSources/05_PaymentResponseIdealSource.yaml [hunk 9]
```diff
@@ -1,4 +1,26 @@
...
+      iban:
+        maxLength: 34
+        type: string
+        description: "The IBAN of the Consumer Bank account used for payment.  If governing law prevents Issuers outside the Netherlands  from disclosing this information, field may be omitted.\n"
```
Comment (philippe-leonhardt-cko) [inline spec/components/schemas/Payments/ResponseSources/05_PaymentResponseIdealSource.yaml]: remove extra whitespace

File: spec/components/schemas/Payments/ResponseSources/05_PaymentResponseIdealSource.yaml [hunk 10]
```diff
@@ -1,4 +1,26 @@
...
+      account_holder:
+        type: string
+        description: "Name of the Consumer according to the name of the account used for payment.  In the exceptional case that the consumerName cannot be retrieved by the Issuer,  this is filled with 'N/A'.  If governing law prevents Issuers outside the Netherlands from disclosing this information, field may be omitted.\n"
```
Comment (philippe-leonhardt-cko) [inline spec/components/schemas/Payments/ResponseSources/05_PaymentResponseIdealSource.yaml]: remove extra whitespace

File: spec/components/schemas/Payments/RequestSources/14_PaymentRequestKlarnaSource.yaml (hunk 1 above)
Comment (kevin-reece-cko) [reply spec/components/schemas/Payments/RequestSources/14_PaymentRequestKlarnaSource.yaml]: This seems to be a quirk of differences in docs generation for single line vs multi-line strings. But as this is an auto-generated file I think it's okay to overlook it... given the rendering is the same

Comment (philippe-leonhardt-cko) [reply spec/components/schemas/Payments/RequestSources/14_PaymentRequestKlarnaSource.yaml]: Fine by me

File: spec/components/schemas/Payments/RequestSources/09_PaymentRequestIdealSource.yaml (hunk 6 above)
Comment (kevin-reece-cko) [reply spec/components/schemas/Payments/RequestSources/09_PaymentRequestIdealSource.yaml]: @philippe-leonhardt-cko These double spaces don't actually render on the UI, so I think I'll leave them for now, unless you have strong objections

**Pull request merged**